bin/node/src/commands/net.rs

```diff
@@ -54,10 +54,21 @@ impl NetCommand {
         // Setup the RPC server with the P2P RPC Module
         let (tx, rx) = tokio::sync::mpsc::channel(1024);
         let p2p_module = NetworkRpc::new(tx.clone()).into_rpc();
-        let rpc_config = RpcConfig::from(&self.rpc);
-        let mut launcher = rpc_config.as_launcher().merge(Some(p2p_module))?;
-        let handle = launcher.launch().await?;
-        info!(target: "net", "Started RPC server on {:?}:{}", rpc_config.listen_addr, rpc_config.listen_port);
+        let rpc_config = Option::<RpcConfig>::from(self.rpc);
+
+        let handle = if let Some(rpc_config) = rpc_config {
+            let mut launcher = rpc_config.as_launcher();
+            launcher.merge(p2p_module)?;
+
+            info!(target: "net", "Starting RPC server on {:?}", launcher.socket());
+
+            let handle = launcher.launch().await?;
+
+            Some(handle)
+        } else {
+            info!(target: "net", "RPC server disabled");
+            None
+        };
```

Comment:
This feels more complex to me than the previous interface. Is the goal of this to hoist the rpc enabled/disabled branch logic?